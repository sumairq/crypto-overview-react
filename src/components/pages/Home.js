import React, { useState } from 'react';
import { useSelector } from 'react-redux';
import Coins from '../Coins';
import Search from '../Search';

const Home = () => {
  const coinsList = useSelector((state) => state.coins);
  const [search, setSearch] = useState('');
  const handleChange = (e) => {
    setSearch(e.target.value);
  };

  const filteredCoins = coinsList.filter((coin) => coin.name.toLowerCase()
    .includes(search.toLowerCase()));
  return (
    <div>
      <Search handleChange={handleChange} />
      <Coins coinsList={filteredCoins} />
    </div>
  );
};

export default Home;
