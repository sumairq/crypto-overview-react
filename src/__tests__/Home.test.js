import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import Home from '../components/pages/Home';
import store from '../redux/configureStore';

test('matches home component snapshot', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <Home />
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
