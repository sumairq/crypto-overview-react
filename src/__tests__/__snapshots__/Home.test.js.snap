// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches home component snapshot 1`] = `
<div>
  <div
    className="search"
  >
    <form>
      <input
        onChange={[Function]}
        placeholder="Search"
        type="text"
      />
    </form>
  </div>
  <p
    className="txt-1"
  >
    Showing 100 most active cryptocurrencies
  </p>
  <div
    className="coins"
  />
</div>
`;
